### ISSUE TITLE
utils.namify does not handle http://localhost:8082/stubs/webapi/first_page

### ISSUE THREAD

**Issue by dudycooly** [deontologician/restnavigator#10]
```
restnavigator/utils.py 

def namify(root_uri):
    '''Turns a root uri into a less noisy representation that will probably
    make sense in most circumstances. Used by Navigator's __repr__, but can be
    overridden if the Navigator is created with a 'name' parameter.'''

    root_uri = unidecode.unidecode(urllib.unquote(root_uri).decode('utf-8'))
    print("Root URI",root_uri)

    generic_domains = set(['herokuapp', 'appspot'])
    print("GENERIC_DOMAINS",generic_domains)
    urlp = urlparse.urlparse(fix_scheme(root_uri))
    print("URLP",urlp)
    formatargs = collections.defaultdict(list)
    print("FORMATARGS",formatargs)

    print("RSPLIT O/P",urlp.netloc.lower().rsplit('.', 1))
    domain, tld = urlp.netloc.lower().rsplit('.', 1)
```

```
In [8]:

root_uri = 'http://localhost:8082/stubs/webapi/first_page'
namify(root_uri)
```

```
---------------------------------------------------------------------------
Root URI http://localhost:8082/stubs/webapi/first_page
GENERIC_DOMAINS set(['herokuapp', 'appspot'])
URLP ParseResult(scheme='http', netloc='localhost:8082', path='/stubs/webapi/first_page', params='', query='', fragment='')
FORMATARGS defaultdict(<type 'list'>, {})
RSPLIT O/P ['localhost:8082']

ValueError: need more than 1 value to unpack

ValueError                                Traceback (most recent call last)
<ipython-input-8-5fe46adc3757> in <module>()
      1 root_uri = 'http://localhost:8082/stubs/webapi/first_page'
----> 2 namify(root_uri)

<ipython-input-7-c3e26188bdd3> in namify(root_uri)
     16 
     17     print("RSPLIT O/P",urlp.netloc.lower().rsplit('.', 1))
---> 18     domain, tld = urlp.netloc.lower().rsplit('.', 1)



```

**Comment by deontologician** [comment]
Ah, ignoring the port probably. Should have thought of that

**Comment by dudycooly** [comment]
It would fail without a port as well, because the line always expect tld in the uri

domain, tld = urlp.netloc.lower().rsplit('.', 1)

And I noticed, the namify() is not returning any value,  the following line in _init_ doesnt have any effect ?!  I guess, we can turn this off for now

```
def __init__(self, root,
                 apiname=None,
                 auth=None,
                 headers=None,
                 session=None,
                 cache=False,
                 curie=None,
    ):
        self.root = utils.fix_scheme(root)
       self.apiname = utils.namify(root) if apiname is None else apiname
```

**Comment by deontologician** [comment]
We only use namify if the user doesn't specify their own name.

**Comment by deontologician** [comment]
@dudycooly there is a fix up in #12 for this. Could you see if this fixes your use case?

**Comment by dudycooly** [comment]
It doesnt handle ip address in url.  Added support for those scenarios and created PR  to namify_broken. let me know if you need it for next

**Comment by deontologician** [comment]
Cool, everything fixed. Merged in #12
